src/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -149,10 +149,58 @@ public static void CreateUncompressedArchive()
             }
         }
 
+        [Fact]
+        public static void CreateNormal_VerifyDataDescriptor()
+        {
+            using var memoryStream = new MemoryStream();
+            // We need an non-seekable stream so the data descriptor bit is turned on when saving
+            var wrappedStream = new WrappedStream(memoryStream, true, true, false, null);
+
+            // Creation will go through the path that sets the data descriptor bit when the stream is unseekable
+            using (var archive = new ZipArchive(wrappedStream, ZipArchiveMode.Create))
+            {
+                CreateEntry(archive, "A", "xxx");
+                CreateEntry(archive, "B", "yyy");
+            }
+
+            AssertDataDescriptor(memoryStream, true);
+
+            // Update should flip the data descriptor bit to zero on save
+            using (var archive = new ZipArchive(memoryStream, ZipArchiveMode.Update))
+            {
+                ZipArchiveEntry entry = archive.Entries[0];
+                using Stream entryStream = entry.Open();
+                StreamReader reader = new StreamReader(entryStream);
+                string content = reader.ReadToEnd();
```

Comment:
nit: you read the stream, didn't do anything with the output, and also seeked to the end (which I believe is a noop if you read to the end already).  Maybe just delete this code that's using the reader and just keep the seek call.